extensions-core/azure-extensions/pom.xml

```diff
@@ -151,7 +151,7 @@
             <groupId>org.apache.druid</groupId>
             <artifactId>druid-server</artifactId>
             <version>${project.parent.version}</version>
-            <scope>test</scope>
+            <scope>compile</scope>
```

Comment:
Should move it up with rest of the Druid's dependencies. 

`compile` seems incorrect, I think it should be `provided` ?